Kitodo/src/main/java/org/kitodo/production/KitodoProduction.java

```diff
@@ -0,0 +1,157 @@
+ * supports the digitization process of various types of materials.
+ */
+@WebListener
+public class KitodoProduction implements ServletContextListener, HttpSessionListener {
+    private static final Logger logger = LogManager.getLogger(KitodoProduction.class);
+    private static CompletableFuture<KitodoProduction> instance = new CompletableFuture<>();
+    private ServletContext context;
+    private Optional<Manifest> manifest;
+    private KitodoVersion version = new KitodoVersion();
+    private ActiveMQDirector activeMQDirector;
+
+    /* This is the main() function. The class is instantiated by the servlet
+     * container, which then calls this function. */
+    @Override
+    public void contextInitialized(ServletContextEvent event) {
+        context = event.getServletContext();
+        manifest = retrieveManifestFileAsStream(context);
+        manifest.ifPresent(version::setupFromManifest);
+        instance.complete(this);
+        startActiveMQ();
+    }
+
+    private static final Optional<Manifest> retrieveManifestFileAsStream(ServletContext context) {
+        try (InputStream manifestResource = context.getResourceAsStream("/META-INF/MANIFEST.MF")) {
+            if (Objects.nonNull(manifestResource)) {
+                return Optional.of(new Manifest(manifestResource));
+            }
+        } catch (IOException e) {
+            logger.error(e.getMessage(), e);
+            context.log(e.getMessage());
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Start up the active MQ connection.
+     */
+    private void startActiveMQ() {
+        if (ConfigCore.getOptionalString(ParameterCore.ACTIVE_MQ_HOST_URL).isPresent()) {
+            activeMQDirector = new ActiveMQDirector();
+            Thread connectAsynchronously = new Thread(activeMQDirector);
+            connectAsynchronously.setName(ActiveMQDirector.class.getSimpleName());
+            connectAsynchronously.setDaemon(true);
+            connectAsynchronously.start();
+        }
+    }
+
+    /**
+     * Returns the application’s main class.
+     * 
+     * @return the main class
+     * @throws UndeclaredThrowableException
+     *             if the servlet container is shut down before the web
+     *             application is started
+     */
+    public static KitodoProduction getInstance() {
+        try {
+            return instance.get();
+        } catch (InterruptedException | ExecutionException e) {
+            logger.fatal(e);
+            throw new UndeclaredThrowableException(e);
+        }
+    }
+
+    /**
+     * Returns the servlet context.
+     * 
+     * @return the servlet context
+     */
+    public ServletContext getServletContext() {
+        return context;
+    }
+
+    /**
+     * Returns the manifest. If there is one. Otherwise, it can also be empty.
+     * 
+     * @return the manifest
+     */
+    public Optional<Manifest> getManifest() {
```

Comment:
See above.

Reply:
It is my idea that the main class of the application also provides essential characteristics of the application via getters. The manifest is such.

Reply:
Please remove unused getters if they are not technically required, see above.